soc/espressif/esp32c2/soc.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2024 Espressif Systems (Shanghai) Co., Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __SOC_H__
+#define __SOC_H__
+
+#ifndef _ASMLANGUAGE
+#include <soc/soc.h>
+#include <rom/ets_sys.h>
+#include <rom/spi_flash.h>
+#include <zephyr/types.h>
+#include <stdbool.h>
+#include <esp_private/esp_clk.h>
+#endif
+
+#ifndef _ASMLANGUAGE
+
+void __esp_platform_start(void);
+
+static inline uint32_t esp_core_id(void)
```

Comment:
Can we remove this? Either use Zephyr's arch API or cpu_ll.h if needed.

Reply:
This is used in a couple of drivers and is present in all other SoC's.
Should we do a separate PR?